src/board/v3/nap/track_channel.c

```diff
@@ -50,45 +50,177 @@ static struct {
   u32 len;
 } nap_ch_state[NAP_MAX_N_TRACK_CHANNELS];
 
-static u32 calc_length_samples(u8 codes, s32 cp_start, u32 cp_rate)
+/** Compute the correlation length in the units of sampling frequency samples.
+ * \param chips_to_correlate The number of chips to correlate over.
+ * \param cp_start Initial code phase in NAP units.
+ * \param cp_rate Code phase rate.
+ * \return The correlation length in NAP units
+ */
+static u32 calc_length_samples(u32 chips_to_correlate, s32 cp_start, u32 cp_rate)
 {
-  u16 chips = codes * 1023;
-  u64 cp_units = (1ULL << NAP_TRACK_CODE_PHASE_FRACTIONAL_WIDTH) * chips
-                 - cp_start;
+  u64 cp_units = (1ULL << NAP_TRACK_CODE_PHASE_FRACTIONAL_WIDTH) *
+                 chips_to_correlate - cp_start;
   u32 samples = cp_units / cp_rate;
   return samples;
 }
 
+/** Return the init value of NAP G1 pseudorandom generator.
+ * \param sid Signal ID
+ * \return NAP G1 initial value
+ */
+u32 sid_to_init_g1(gnss_signal_t sid)
```

Comment:
This function is specific to NAP v3. As this file is in v3/nap directory, it looks to me that it belongs here quite naturally. I could not find a better place for this function in libswiftnav. Anyways, if you could pinpoint the place in libswiftnav where I could move this function, then I will move it there.